xblock/test/test_fields.py

```diff
@@ -780,7 +810,11 @@ def test_to_string_regexp_matches(self, _type, value, regexp):
                   kaw: null
             """),
             [1, 2.345, {"foo": True, "bar": [1, 2, 3]}, {"meow": False, "woof": True, "kaw": None}]
-        )
+        ),
+        # Test that legacy DateTime format including double quotes can still be imported for compatibility with
+        # old data export tar balls.
+        (DateTime, '"2014-04-01T02:03:04.567890"', dt.datetime(2014, 4, 1, 2, 3, 4, 567890).replace(tzinfo=pytz.utc)),
+        (DateTime, '"2014-04-01T02:03:04.000000"', dt.datetime(2014, 4, 1, 2, 3, 4).replace(tzinfo=pytz.utc)),
     )
```

Comment:
@doctoryes These two functions essentially test the same thing, yes.  The tests are structured in two groups since some of them work both ways, whereas others only test deserialisation of strings that we should be able to deserialise, but that would not be serialised in exactly the way specified in the test data.  All test data for `test_both_directions()` falls into the first category, and all test data for `test_from_string()` falls into the second category, for various reasons.  Some of them test YAML code that isn't JSON, others test non-standard capitalisation or number formats.

I added the `DateTime` tests to the two functions according to the same principles:  The ones that are symmetric go in the first function, the ones that only test correct deserialisation go in the second function.  Maybe I could add a comment to make the idea more obvious?